### DISCUSSION TITLE
App reloads every minute

### DISCUSSION THREAD

**Discussion by kemeris2000** [lowdefy/lowdefy#1605] [Questions and Help]
My app reloads automatically approx. every minute without filesystem modifications for user side. I suspect it is because process opening and closing without write two files: 
```
app/.lowdefy/dev/build/global.json
app/.lowdefy/dev/build/menus.json
```

![image](https://github.com/lowdefy/lowdefy/assets/48207709/8c51a6e4-e3f5-4173-ae1d-4100bac33f41)

I am running my v4.0.0-rc.9 app with:
`pnpm lowdefy dev `

Any ideas how to solve this problem?

**Comment by Gervwyk** [comment]
Hi @kemeris2000 this is just the dev server that watches files for configuration changes. Seems like you might have restricted directory access to the .lowdefy folder, so the dev server cannot rewrites these files and incrementally retries. Check that the dev server process has read and write access to the .lowdefy folder. 

There is no watcher for the production app. So you can also use `pnpm lowdefy build && pnpm lowdefy start` and it should build and startup the lowdefy production server.

**Comment by kemeris2000** [comment]
Looks like it's not a permission problem as processes running under root account.

<pre><code>
root@www-dev:/var/www/example.com# ps -aux | grep -E 'lowdefy|pnpm|node'
root      376517  0.1  4.6 880480 91472 pts/5    Sl+  22:32   0:01 node /usr/bin/pnpm lowdefy dev
root      376528  0.1  3.5 21690540 69420 pts/5  Sl+  22:32   0:00 node /var/www/example.com/app/node_modules/.bin/../../../node_modules/.pnpm/lowdefy@4.0.0-rc.9/node_modules/lowdefy/dist/index.js dev
root      376567  0.1  5.1 886168 101068 pts/5   Sl+  22:32   0:01 node /usr/bin/pnpm run start
root      376578  0.0  0.0   2488   492 pts/5    S+   22:32   0:00 sh -c node manager/run.mjs
root      376579  0.4  3.8 11341784 75832 pts/5  Sl+  22:32   0:03 node manager/run.mjs
root      376640  0.1  4.2 11155668 84636 pts/5  Sl+  22:32   0:01 node /var/www/example.com/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/bin/next start
root@www-dev:/var/www/example.com#
root@www-dev:/var/www/example.com# ls -al app | grep ".lowdefy"
drwxr-xr-x 4 www-data www-data 4096 Jun 27 22:14 .lowdefy
-rw-r--r-- 1 www-data www-data 8408 Jun 26 22:02 lowdefy.yaml
root@www-dev:/var/www/example.com#
root@www-dev:/var/www/example.com# echo "test" >> app/.lowdefy/dev/build/global.json
root@www-dev:/var/www/example.com#
root@www-dev:/var/www/example.com# tail -n 5 app/.lowdefy/dev/build/global.json
    "~k": "29y"
  },
  "~k": "29u"
}
test
</pre></code>

On every reload in browser console I see this:

![image](https://github.com/lowdefy/lowdefy/assets/48207709/50e02829-e06a-4bc4-aea5-824cd657afd8)

However production server works and it's acceptable for me. Thank you for suggestions.

**Comment by kemeris2000** [comment]
This problem disappeared after update to v4.0.0-rc.12.
